contracts/contracts/coordination/Coordinator.sol

```diff
@@ -65,6 +65,8 @@ contract Coordinator is AccessControlDefaultAdminRules {
 
     bytes32 public constant INITIATOR_ROLE = keccak256("INITIATOR_ROLE");
 
+    mapping(address => BLS12381.G2Point) public providerPublicKey;
```

Comment:
I'm out of context, why here provider used instead of operator?

Reply:
Because the mapping key should be the provider not the operator (although technically the key is generated by the node)

Reply:
That's my question why provider key? Provider is technically staker and I wonder why staker needed

Reply:
We keep the model where the node identifier from a protocol point of view is the staker (staking provider if we're exact), similar to how NuCypher works.

Reply:
Perhaps this should be plural `providerPublicKeys` instead of `providerPublicKey`?

I guess presumably the expectation is that on start-up, Ursulas would need to check that they have already submitted the relevant public key or update it if not? This would be before starting to process any ritual events. (cc @KPrasch )

Reply:
Documented downstream changs in `nucypher` here: https://github.com/nucypher/nucypher/issues/3167

Reply:
> Perhaps this should be plural `providerPublicKeys` instead of `providerPublicKey`?

I don't know, I personally tend to use singular for arrays and mappings. But I'm fine with adopting a common convention for this.